migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/migration_impl.go

```diff
@@ -0,0 +1,146 @@
+	oidcAuthProviders := make([]*storage.AuthProvider, 0, batchSize)
+
+	err := authProviderStore.Walk(ctx, func(obj *storage.AuthProvider) error {
+		if obj.GetType() == oidc.TypeName {
+			oidcAuthProviders = append(oidcAuthProviders, obj)
+			if len(oidcAuthProviders) == batchSize {
+				if err := migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders); err != nil {
+					return err
+				}
+				oidcAuthProviders = oidcAuthProviders[:0]
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders)
+}
+
+func migrateAuthProviderClaims(ctx context.Context, groupStore groups.Store, providerStore authproviders.Store, providers []*storage.AuthProvider) error {
+	providerToGroups := make(map[string][]*storage.Group, len(providers))
+	for _, provider := range providers {
+		providerToGroups[provider.GetId()] = make([]*storage.Group, 0)
+	}
+	err := groupStore.Walk(ctx, func(obj *storage.Group) error {
+		authProviderID := obj.GetProps().GetAuthProviderId()
+		if _, ok := providerToGroups[authProviderID]; ok {
+			providerToGroups[authProviderID] = append(providerToGroups[authProviderID], obj)
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to get groups from the store")
+	}
+	migratedProviders := make([]*storage.AuthProvider, 0)
+	migratedGroups := make([]*storage.Group, 0)
+	for _, provider := range providers {
+		fixOrgID := (hasRequiredAttribute(orgIDKey, provider) || hasGroupKey(orgIDKey, providerToGroups[provider.GetId()])) && !hasClaimMapping(accountIDClaim, orgIDKey, provider)
+		if fixOrgID {
+			setClaimMapping(provider, accountIDClaim, orgIDKey)
+		}
+		fixIsOrgAdmin := false
+		for _, requiredAttribute := range provider.GetRequiredAttributes() {
+			if requiredAttribute.GetAttributeKey() == groupsKey && requiredAttribute.GetAttributeValue() == orgAdminGroupsValue {
+				setClaimMapping(provider, isOrgAdminKey, orgAdminGroupsValue)
+				requiredAttribute.AttributeKey = orgAdminGroupsValue
+				requiredAttribute.AttributeValue = "true"
+				fixIsOrgAdmin = true
+			}
+		}
+		for _, group := range providerToGroups[provider.GetId()] {
+			if group.GetProps().GetKey() == groupsKey && group.GetProps().GetValue() == orgAdminGroupsValue {
+				setClaimMapping(provider, isOrgAdminKey, orgAdminGroupsValue)
+				group.GetProps().Key = orgAdminGroupsValue
+				group.GetProps().Value = "true"
+				migratedGroups = append(migratedGroups, group)
+				fixIsOrgAdmin = true
+			}
+		}
+
+		if fixOrgID || fixIsOrgAdmin {
+			migratedProviders = append(migratedProviders, provider)
+			if provider.GetTraits().GetOrigin() != storage.Traits_IMPERATIVE {
+				log.Warnf("Declarative auth provider with id %s and name %s uses claims that were removed from the list of default claims. "+
+					"Please add claim mapping for account_id or is_org_admin if you are using any of these claims", provider.GetId(), provider.GetName())
```

Comment:
There was precedent of another migration doing a log statement which required actions from the user: https://github.com/stackrox/stackrox/blob/master/migrator/migrations/m_171_to_m_172_move_scope_to_collection_in_report_configurations/migration.go#L253-L256

Might be worthwhile to switch this to an error log for more "visibility" and share details? Like we could even give them the YAML for the claim mappings as well if we really wanted to.